### PR TITLE
Numeracy Programming Challenges

### PR DESCRIPTION
**Pull request by ravenmaster001**
Addresses #247 

Looks as though this branch claimed the history of `milestone/unplugged-programming` :\

### CODE REVIEW

**Comment by ravenmaster001** [comment]
@JackMorganNZ Uh oh ignore this for now something went wrong with the last merge because of the history voodoo trickery.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/add-in-parts-1-digit-number/add-in-parts-1-digit-number.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    <td>9<br>3</td>
+    <td>(9 + 1) + 2 = 12</td>
+  </tr>
+  <tr>
+    <td>5<br>4</td>
+    <td>5 + 4 = 9</td>
+  </tr>
+</table>
+
+Click on the green flag, enter the inputs provided in the “testing examples” to see the expected output of your program.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/add-in-parts-1-digit-number/add-in-parts-1-digit-number.md]: This line should be deleted, it's already in the `scratch-expected` section.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/add-in-parts-1-digit-number/scratch-hints.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-  Make two variables called “num1” and “num2” and set their values to the inputs entered by the user.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/add-in-parts-1-digit-number/scratch-hints.md]: This list is missing a space after each bullet point.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/add-in-parts-2-digit-number/scratch-hints.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+-  Make two variables called “num1” and “num2” and set their values to the inputs entered by the user.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/add-in-parts-2-digit-number/scratch-hints.md]: This list is missing a space after each bullet point.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/calculate-exponents/calculate-exponents.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <td>3^5 = 243</td>
+  </tr>
+  <tr>
+    <td>-2<br>4</td>
+    <td>-2^4 = 16</td>
+  </tr>
+</table>
+
+
+Click on the green flag, enter the inputs provided in the “testing examples” to see the expected output of your program.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/calculate-exponents/calculate-exponents.md]: This line should not be displayed in the description section, as it's Scratch dependent.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/calculate-exponents/scratch-hints.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+-  Make a variable called “base” and set its value to the base number entered by the user. This is the number that will be multiplied by.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/calculate-exponents/scratch-hints.md]: This list is missing a space after each bullet point.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/check-if-divisor-series/scratch-hints.md [hunk 6]
```diff
@@ -0,0 +1 @@
+-  Divisor is a number that divides into another without a remainder (i.e. a number modulo its divisor is 0). You can find the `scratch:() mod ()` block under the “operators” script.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/check-if-divisor-series/scratch-hints.md]: This list is missing a space after each bullet point.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/check-if-divisor/scratch-hints.md [hunk 7]
```diff
@@ -0,0 +1 @@
+-  Divisor is a number that divides into another without a remainder (i.e. a number modulo its divisor is 0). You can find the `scratch:() mod ()` block under the “operators” script
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/check-if-divisor/scratch-hints.md]: This list is missing a space after each bullet point.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/find-factors/scratch-hints.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+We have provided 2 different algorithms to code this program;
+
+Algorithms to find the factors of “number”:
+
+Solution 1:
+
+1. Create a variable called “divisor” and set its value to 1.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/find-factors/scratch-hints.md]: This list is missing a space after each number.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/find-prime-factors/scratch-hints.md [hunk 9]
```diff
@@ -0,0 +1 @@
+-  You need to find the factors of the number (see challenge 4.1) and check each factor to see if it’s a prime number (see challenge 4.2). Add the factors which are also prime numbers to a string variable called “prime factors”.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/find-prime-factors/scratch-hints.md]: This list is missing a space after each bullet point.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/flip-coins/scratch-hints.md [hunk 10]
```diff
@@ -0,0 +1,2 @@
+-  To simulate a coin toss, use the block `scratch: pick random (1) to (10)` and replace the number 10 with the number 2 (generating random numbers ‘1’ for tail, or ‘2’ for head).
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/flip-coins/scratch-hints.md]: This list is missing a space after each bullet point.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/generate-primes/scratch-hints.md [hunk 11]
```diff
@@ -0,0 +1 @@
+-  To generate prime numbers up to a number, check every number less than the number (one at a time) to find out if it’s prime or not. If it’s a prime number, add it to a string variable called “prime numbers” and check the next number.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/generate-primes/scratch-hints.md]: This list is missing a space after each bullet point.

File: csunplugged/topics/content/en/unplugged-programming/programming-challenges/go-up-1-10-100-numbers/scratch-hints.md [hunk 12]
```diff
@@ -0,0 +1,5 @@
+-  Make a list of length 3 containing numbers 1, 10 and 100.
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/unplugged-programming/programming-challenges/go-up-1-10-100-numbers/scratch-hints.md]: This list is missing a space after each bullet point.

**Comment by ravenmaster001** [comment]
I also fixed all the red scratch blocks within my exercise additions.

**Comment by ravenmaster001** [comment]
Additional Fixes:

- purple scratch blocks are now purple
- updated all other programming markdown files in `unplugged-programming` for consistency:
  + sentences are now separated by newlines
  + language independence in challenge descriptions
  + list spacing

**Pull request merged**